crates/ruff_python_formatter/README.md

```diff
@@ -3,10 +3,205 @@
 The goal of our formatter is to be compatible with Black except for rare edge cases (mostly
 involving comment placement).
 
-## Implementing a node
+## Dev tools
 
-Formatting each node follows roughly the same structure. We start with a `Format{{Node}}` struct
-that implements Default (and `AsFormat`/`IntoFormat` impls in `generated.rs`, see orphan rules below).
+**Testing your changes** You can use the `ruff_python_formatter` binary to format individual files
+and show debug info. It's fast to compile because it doesn't depend on `ruff`. The easiest way is to
+create a `scratch.py` (or `scratch.pyi`) in the project root and run
+
+```shell
+cargo run --bin ruff_python_formatter -- --emit stdout scratch.py
+```
+
+which has `--print-ir` and `--print-comments` options. We especially recommend `--print-comments`.
+
+<details>
+<summary>Usage example</summary>
+
+Command
+
+```shell
+cargo run --bin ruff_python_formatter -- --emit stdout --print-comments --print-ir scratch.py
+```
+
+Input
+
+```python
+def f():  # a
+    pass
+```
+
+Output
+
+```text
+[
+  "def f",
+  group([group(["()"]), source_position(7)]),
+  ":",
+  line_suffix(["  # a"]),
+  expand_parent,
+  indent([hard_line_break, "pass", source_position(21)]),
+  hard_line_break,
+  source_position(21),
+  hard_line_break,
+  source_position(22)
+]
+{
+    Node {
+        kind: StmtFunctionDef,
+        range: 0..21,
+        source: `def f(): # a⏎`,
+    }: {
+        "leading": [],
+        "dangling": [
+            SourceComment {
+                text: "# a",
+                position: EndOfLine,
+                formatted: true,
+            },
+        ],
+        "trailing": [],
+    },
+}
+def f():  # a
+    pass
+```
+
+</details>
+
+The other option is to use the playground (also check the playground README):
+
+```shell
+cd playground && npm run dev:wasm && npm install && npm run dev
```

Comment:
```suggestion
cd playground && npm install && npm run dev:wasm && npm run dev
```

Reply:
fwiw the playground readme is still the other way round